### PR TITLE
feat: Add col_count, row_count and is_empty methods

### PR DESCRIPTION
**Pull request by Techassi**
This PR adds three methods:

- `column_count`: Returns the number of columns in the table
- `row_count`: Returns the number of rows in the table
- `is_empty`: Returns if the table is empty (contains no data rows)

### CODE REVIEW

File: src/table.rs [hunk 1]
```diff
@@ -118,6 +118,20 @@ impl Table {
...
+    /// use comfy_table::Table;
+    ///
+    /// let mut table = Table::new();
+    /// table.set_header(vec!["Col 1", "Col 2", "Col 3"]);
+    ///
+    /// assert_eq!(table.col_count(), 3);
+    /// ```
+    pub fn col_count(&self) -> usize {
+        self.columns.len()
+    }
```
Comment (Nukesor) [inline src/table.rs:133]: This is not always accurate though. See `discover_columns`.
Also, we should stick to the naming convention of the table and don't use any abbreviations. -> `column_count`.

**Review by Nukesor — CHANGES_REQUESTED** [review]
I actually want to challenge the addition of the `_count` methods.

I'm not sure if this is particularily useful, since the alternative `.row_iter().count()` and `.column_iter().count()` really isn't that much longer and more complex.
I don't see the benefit of adding this yet.

Regarding the `is_empty()` call; I'm a bit more inclined to do this, since the alternative would be `row_iter().next().is_none()`, which isn't ergonomic at all.

But I want to ponder for a moment on what an empty table actually is?
Is it a table without rows?
What if there's a header?
What if there're rows, but the rows don't have any cells?

Personally I'm not sure which of these conditions should be met for a table to be considered empty.

**Comment by Techassi** [comment]
> I actually want to challenge the addition of the _count methods.

> I'm not sure if this is particularily useful, since the alternative .row_iter().count() and .column_iter().count() really isn't that much longer and more complex.
I don't see the benefit of adding this yet.

The current implementation avoids constructing an iterator which is then immediately consumed by the `.count()` call.

> But I want to ponder for a moment on what an empty table actually is?
> Is it a table without rows?
> What if there's a header?
> What if there're rows, but the rows don't have any cells?

Yes I was thinking about the same stuff you did. Currently I consider the table to be empty if there are no *data* rows. This excludes the header. A header being there without any *data* rows is still considered empty.

We could expand this implementation and also check if the rows contain cells, and if they don't, the table will be empty.

File: src/table.rs (hunk 1 above)
Comment (Techassi) [reply src/table.rs:133]: > This is not always accurate though. See discover_columns.

Ah yes, you are right. We could make sure to call `discover_columns` before returning the length.

> Also, we should stick to the naming convention of the table and don't use any abbreviations. -> column_count.

Yes, agreed. I will change that!

Comment (Techassi) [reply src/table.rs]: Renamed in 044cf03. I also added the `discover_columns` call.

**Comment by Techassi** [comment]
Any update on this? @Nukesor

**Comment by Nukesor** [comment]
I guess these changes are fine :) 

The argument of "a table is empty if no data rows are in it" seems reasonable as well.

Thanks for the contribution!

**Review by Nukesor — APPROVED** [review]

**Pull request merged**